### PR TITLE
Revised import flow (WIP) (close #3)

### LINKED ISSUE #3: Improve import flow
The current import flow is tedious and manual. The user has to select a specific file or folder that exactly matches what their data source expects. It would be much better if you could select a folder (or use the default; say the home folder) and the app could recursively scan the folder for files/folders it recognizes. "Import everything recognized from my Home folder."

I'm well into this huge refactor in a yet-unpublished branch, and it's actually quite tricky to do this efficiently.

There are two main types of data sources: those which import entire folders/file trees, and those that import individual files, but _may_ (not necessarily) also import an entire folder/file tree if it consists mostly of that type of file. (Note that archive files can be treated the same as folders, as the app can traverse into archive files as if they were folders.)

For example, the Twitter/X data source imports an entire folder (or archive) because it expects a certain structure from the Twitter account export. There are certain files in certain places and all it has to do is check them; but it's very fast and covers an entire folder tree. No more traversing that folder is needed once recognized.

On the other hand, some data sources like the Media (photos/videos/audio) or VCard importers work with a single file. (Media actually also works with sidecar files, which it may check for adjacent to an imported media file; this is usually a metadata file or a motion picture file.) But these can also import an entire folder of photos or VCards if a folder consists primarily of those files. So, these data sources also traverse folders on their call to `Recognize()`.

This is inefficient, since this traverses the file system multiple times, and is indeed quite slow...

Ideally, the data sources which import folders will recognize their folders, and data sources which import files will just recognize their files, and the importer logic will figure out whether to import the entire folder or not.

As part of this refactor, it's time to iron out how imports are represented in the database, since this is a good time to figure out how checkpointing and resuming works. (Checkpointing/resuming is a feature that was previously supported way back in Photobak and Timeliner, when importing data directly via APIs.)

Since this is such a large change, I don't expect I'll commit much or accept (m)any contributions to the main branch until it is done, but then we can roll forward again.

### PR DESCRIPTION
**Pull request by mholt**
This change will touch on quite a few fundamental aspects of the program (thumbnails, jobs/tasks, logging, configuration, etc), so quite a big change. Hopefully one of the last big ones for a while.

Very WIP right now. Very broken :)

### CODE REVIEW

File: timeline/schema.sql [hunk 1]
```diff
@@ -361,6 +362,23 @@ CREATE TABLE IF NOT EXISTS "review_codes" (
...
+	"title" TEXT,
+	"description" TEXT,
+	"type" TEXT,
+	"value",
+	"item_id" INTEGER,
+	"attribute_id" INTEGER,
+	"data_source_id" INTEGER,
+	FOREIGN KEY ("job_id") REFERENCES "jobs"("id") ON UPDATE CASCADE ON DELETE CASCADE,
+	FOREIGN KEY ("item_id") REFERENCES "items"("id") ON UPDATE CASCADE ON DELETE CASCADE,
+	FOREIGN KEY ("entity_id") REFERENCES "entities"("id") ON UPDATE CASCADE ON DELETE CASCADE,
```
Comment (mohammed90) [inline timeline/schema.sql:377]: `job_id` and `entity_id` are not columns in the table 🤔

Comment (mholt) [reply timeline/schema.sql]: Ohai Mohammed! You're right... still WIP, and very very tired commit :sweat_smile:

**Comment by mholt** [comment]
I detoured slightly and also added some ML features into this branch, including image search and similar item search.

**Comment by mholt** [comment]
It's working! Still lots of polish and some bug fixing to do, but the new import page, jobs logic, thumbnail and embeddings system, are all in place!

Lint errors are known, some are false positives -- I'll fix those later.

**Pull request merged**